### DISCUSSION TITLE
Cors errors Nodejs + Angular + Railway Deploy

### DISCUSSION THREAD

**Discussion by torresmald** [socketio/socket.io#4676] [Q&A] [answered]
Hi, Im stuck for several weeks triying socket io works when I deploy the back in cloud.

Here is my back:
const express = require('express');
const cors = require('cors');
const bodyParse = require('body-parser');
const app = express();
const htpp = require('http').Server(app);
const PORT = process.env.PORT || 4000;
const io = require('socket.io')(htpp, {
    cors: {
        origins: ["http://localhost:4200"],
        credentials: true,
        allowedHeaders: ["my-custom-header"],
   }
});
app.use(cors());
app.use(bodyParse.urlencoded({extended: true}))
app.use(bodyParse.json());

io.on('connection', (socket) => {
    console.log('Nuevo usuario Conectado');
    socket.on('sendMessage', (messageInfo) => {
        socket.broadcast.emit('receiveMessage', messageInfo);
    })
})
app.get('/', (request, response) => {
    response.status(200).json('Bienvenido a la API.')
})
htpp.listen(PORT, () => {
    console.log(`Escuchando en http://localhost:${PORT}`);
});
module.exports = app;

Here the Service with Angular: 

export class SocketService {
  io = io("https://socketchatsemilla.up.railway.app/", {
    autoConnect: true,
    withCredentials: true,
    extraHeaders: {
      "my-custom-header": "abcd"
    }
  })
  constructor() { 
  }
}

The errors:

Access to XMLHttpRequest at 'https://socketchatsemilla.up.railway.app/socket.io/?EIO=4&transport=polling&t=OSkgJlF' from origin 'http://localhost:4200' has been blocked by CORS policy: The value of the 'Access-Control-Allow-Origin' header in the response must not be the wildcard '*' when the request's credentials mode is 'include'. The credentials mode of requests initiated by the XMLHttpRequest is controlled by the withCredentials attribute.
zone.js:2710         

 GET https://socketchatsemilla.up.railway.app/socket.io/?EIO=4&transport=polling&t=OSkgJlF net::ERR_FAILED 200

Im literally getting crazy. Tried with Vercel and the same.

**Comment by darrachequesne** [comment, marked as answer]
Hi! I think it's `origin`, not `origins` in the `cors` option:

```js
const io = require('socket.io')(htpp, {
  cors: {
    origin: ["http://localhost:4200"],
    credentials: true,
    allowedHeaders: ["my-custom-header"],
  }
});
```

Reference: https://socket.io/docs/v4/server-options/#cors

**Reply by torresmald** [reply]
Hi! Sorry i didn´t answer but i was full with the Final Project delivery.  Finally i solved using Railway instead Vercel... was so quick the solution...
